### DISCUSSION TITLE
From Wallet to Everything

### DISCUSSION THREAD

**Discussion by hanbu97** [creatorsand/.github#13] [Ideas]
**整体的想法是以一个钱包为基础，把本次课程所学尽可能囊括在内，把本次共学的其他项目融合进去**。

### 需要/可以做的事
1. 钱包开发，跑通转账，dapp
2. 合约交互：例如多签，ido等功能
3. 本次共学项目
  a. 社交功能(https://github.com/CreatorsDAO/all-in-one-solana/discussions/19)
  b. zk的身份体系(https://github.com/CreatorsDAO/all-in-one-solana/discussions/22)
  c. 基于eightfish的应用(https://github.com/CreatorsDAO/all-in-one-solana/discussions/18)
  d. 抽象账户体系(https://github.com/CreatorsDAO/all-in-one-solana/discussions/17)
  e. Hackathon/Grants平台(https://github.com/CreatorsDAO/creatorsdao.github.io/discussions/33)
  f. Game Box(https://github.com/CreatorsDAO/creatorsdao.github.io/discussions/37)
...持续迭代

### 技术选型：
- 前端： Flutter
- App后端（本地）：Rust
- Server端（如有必要）：Rust
通过
https://github.com/fzyzcjy/flutter_rust_bridge (已跑通)
https://github.com/cunarist/rust-in-flutter  (可以进行尝试，似乎兼容性有优势)

将rust和flutter进行连接。rust部分直接使用solana官方的sdk，降低开发难度。

现有一个项目使用tdlib连通了telegram，正在此基础上添加钱包功能以及实现基本通信功能。如果与Game Box整合可以带来更加客制化的体验。用wasm + tg webapp + 区块链钱包 这套有望实现微信小程序，小游戏的体验。

### 项目优势
- 此项目的核心竞争力在于和其他的共学项目可以保持紧密的联系。应用需要钱包来链接用户，钱包需要应用来丰富功能。最终实现互惠共赢，达到1+1>2的效果。
- 通过rust和flutter进行桥接，方便在App端进行更多sol操作并且复用现有的代码。

### 联系方式
- Discord(https://discord.gg/9Gt3z6Yv) 中的hanbu97

**Comment by bonnie0519** [comment]
你好呀，想问一下发起人你的社群名

**Reply by hanbu97** [reply]
可以匿名吗，我在Discord上有发言。可以在上面联系。

**Reply by bonnie0519** [reply]
好滴

**Reply by VioletSakura777** [reply]
关注关注，想试试
